[강예지] Refactor: 스스로 리팩토링

Kangyejiii opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
[Refactor] 중복된 코드 Storys Component로 분리/팔로우 추천  Component 로 분리 (8/15)
[Refactor] Login.js 필요없는 div 정리,story mockdata 추가및 적용,중복요소 제거,spread 연산자,시맨틱태그 (8/16) 
		 
## 수정 사항들 자세한 내용
- `import 순서 :  패키지 > 컴포넌트 > 변수/이미지 > 스타일 파일` 순서에 따라  정리했습니다.
- comment.scss 생성해서 스타일을 분리하고, 클래스명을 짧게 고쳐봤습니다.
- 불필요한 console.log() 삭제 
- 팀원들과 공통 스타일이 또 있는지 확인해보고 수정해서 PR 하겠습니다.

 **(8/15)**
- Feeds.js에서 스토리 코드 부분 Component로 분리
- Storys Component 생성
- Storys.scss 생성 ,( Feeds.scss에서 분리 )

**(8/15) 2번째 commit**
- data/yejikangMock 안에 추천리스트 받는 Mockdata 추가 
- fetch 함수가 있는 별도의 파일 추가(RemoteDataSource)
- SuggestionListItem Component 추가
- 관련된 스타일 suggestionListItem. scss 파일 추가 (Aside.scss에서
  스타일 분리)

**(8/15) 3/4번째 commit**

- 비구조화(ES6)를 활용한 코드 수정 

**(8/16) 첫 번째**

- login 레이아웃 정리
- 필요없는 div 제거
- login.scss 수정

**(8/16) 두 번째**

- story mock data 추가 및 적용
- story scroll 숨기기 기능 추가

**(8/16) 세 번째**

- Aside / 하단 도움말: 배열 생성 후 map()
- Feeds / spread 연산자 사용

## 기타 질문 및 특이 사항
- 리팩토링 할 부분이 많아서 commit을 나눠서 하고 있습니다. **(진행중)**  
- 데이터를 가져올 때  fetch 하는 function을 만들어서 import 해봤습니다.(`RemoteDataSource.js`) 이방법은 좋은 방법인가요? 파일 마다 동일하게 작성되는 부분이 될거같아서 분리해봤습니다! 
- 비구조화 할 때 props도 `const {id,image} = this.props` 이렇게 해도 되나요? (`SuggestionListItem.js` 부분 )   

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Kangyejiii committed:
Add : 공통Components  nav bar 추가
- src/Componets/Nav에 생성했습니다.
- js 파일과 scss 파일 2개입니다.
- 로고 이미지는 images/donghokim 으로 경로지정했습니다.
- a 태그가 있습니다. 후에 link 로 수정이 필요할 것같습니다.
- nesting 완료했고 수정할 것이 있으면 알려주세요.

Kangyejiii committed:
[Modify] : 들여쓰기 정리

joonsikyang commented on src/Components/Nav/Nav.js:
```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import "./Nav.scss";
+
+import "../../styles/reset.scss";
+import "../../styles/common.scss";
```
reset.scss, common.scsss는 index.js에서 한 번만 import 해주세요

joonsikyang replied:
수정했습니다.

joonsikyang commented on src/Components/Nav/Nav.js:
```diff
@@ -0,0 +1,82 @@
+import "../../styles/reset.scss";
+import "../../styles/common.scss";
+
+//TODO: a 태그 를 link로 바꿔야합니다. 주의) Nav.scss에 a 태그 스타일이 있음!
```
귀여운 주석이네요🤭  외부 사이트로 연결되는 것이 아니라면 a 태그는 전부 바꿔주세요!

joonsikyang commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,77 @@
+// [yeji] : box-sizing:border-box를 Nav바에다가 적었습니다.
```
box-sizing 속성은 reset.scss 혹은 common.scss로 옮겨주세요

joonsikyang commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,77 @@
+
+	nav {
+		width: 935px;
+		display: flex;
+		justify-content: space-between;
+		align-items: center;
```
이런 속성들은 sass extend 기능을 활용해서 좀 더 효과적으로 다룰 수 있을 것 같네요!

joonsikyang commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,77 @@
+	box-sizing: border-box;
+	width: 100%;
+	height: 54px;
+	display: flex;
+	justify-content: center;
+	position: fixed;
```
Sass 활용 잘해주셨네요🙌 아래 코멘트 확인해주세요

joonsikyang reviewed (commented):
리뷰 반영해서 수정 부탁드립니다. 다시 push 하실 때는 라벨도 달아주세요 :) 

> common.scss에서 mixin 그리고 공통된 color를 변수선언 해서 사용하려고 합니다. 그런데 import를 nav.scss에서 해야만 에러가 뜨지 않습니다...!

공통 변수 사용하려면 원래 .scss 파일에서 import 해서 사용하는게 맞습니다! :)

Kangyejiii committed:
Refactor : 멘토리뷰 반영
- <a> 태그를 Link로 수정
- sass extend 사용
- common.sass에서 지정한 색깔 사용 (ex) $border-grey), mixin 사용
- nav.sass에서 썼던 box-sizing 설정을 지우고 common.css에서 작성

Kangyejiii committed:
Refactor:불필요한 코드제거

joonsikyang commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+}
+
+.Nav {
+	@extend %center-fixed;
```
이러한 속성은 include 키워드가 더 적절합니다. 결과는 같아도 inclue, extend 키워드 잘 구분해서 상황에 맞게 사용해주세요! 아래 링크 참고해보세요.
- https://www.geeksforgeeks.org/include-vs-extend-in-sass/

Kangyejiii replied:
감사합니당😊

joonsikyang commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+	border-bottom: 1px solid $border-grey;
+
+	nav {
+		@include centerSpaceBetween();
```
위 링크 들어가서 mixin 활용하는 부분 봐주세요. 인자를 주는 경우가 아니라면 `centerSpaceBetween` 이렇게만 작성하시면 됩니다.

joonsikyang commented on src/index.js:
```diff
@@ -1,5 +1,7 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import Routes from "Routes";
+import "./styles/reset.scss";
+import "./styles/common.scss";
```
Goood💪

joonsikyang commented on src/styles/common.scss:
```diff
@@ -6,24 +6,27 @@ $dot-grey: rgb(168, 168, 168);
+	justify-content: center;
+	align-items: center;
 }
 
 @mixin startSpaceBetween {
-  display: flex;
-  justify-content: space-between;
+	display: flex;
+	justify-content: space-between;
 }
 
 @mixin centerSpaceBetween {
-  display: flex;
-  justify-content: space-between;
-  align-items: center;
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+}
```
`*`, `body` selector 밑으로 옮겨주세요. 추상적 >>> 구체적 순서로 작성해주시면 좋습니다!

joonsikyang reviewed (commented):
예지님 리뷰 확인 부탁드립니다. 공통 컴포넌트라서 merge가 빨리 되어야 하나요? @Kangyejiii

Kangyejiii committed:
Add : westagram 리액트 파일 세팅
- pages/yejikang 에 Main,Login 파일 추가했습니다.
- public/images/yejikang 에 필요한 이미지 파일 추가했습니다.
- common, reset 이외에 제가 개인적으로 필요한 기본 스타일 세팅을
  base.scss에 작성했습니다. (공통으로 쓰이는 부분이라면  common에
추가하겠습니다)
- 로그인 버튼 조건(1. id에 '@' 포함 2. 비밀번호 5자리 이상)에 따른
  활성화 비활성화 설정
- 댓글 추가는 component로 분리했습니다.
-[Refactor] :  nav Component 수정했습니다.

joonsikyang commented on src/pages/yejikang/Login/Login.js:
```diff
@@ -0,0 +1,88 @@
+  };
+
+  handleSummit = () => {
+    console.log(this.state); //과제 1 ) 아이디 패스워드 로그 출력
```
나중에는 push 할 때 console.log 지워주세요!

joonsikyang commented on src/pages/yejikang/Login/login.scss:
```diff
@@ -0,0 +1,75 @@
+@import "../../../styles/common.scss";
+
+@mixin columnCenter {
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+}
```
이런 속성도 나중에는 common.scss에 지정해서 팀원들과 공유하면 더욱 좋습니다 :)

Kangyejiii replied:
넵!

joonsikyang commented on src/pages/yejikang/Main/Components/Aside/Aside.js:
```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import "./Aside.scss";
+import { Link } from "react-router-dom";
```
import 순서! 패키지 > 컴포넌트 > 변수/이미지 > 스타일 파일

Kangyejiii replied:
import 순서를 신경쓰지 못했네요! 수정하겠습니다~

joonsikyang commented on src/pages/yejikang/Main/Components/Comment/Comment.js:
```diff
@@ -0,0 +1,31 @@
+    };
+  }
+
+  handleClick = () => {
+    this.setState({ isLike: !this.state.isLike });
+  };
```
토글 기능! Good🤘

joonsikyang commented on src/pages/yejikang/Main/Components/Comment/Comment.js:
```diff
@@ -0,0 +1,31 @@
+        <span className="commentId">{this.props.id}</span>
+        {this.props.text}
+        <button
+          className={`commentHeartBtn ${
+            this.state.isLike ? "fillheart" : "heart"
+          }`}
```
Gooood!

joonsikyang commented on src/pages/yejikang/Main/Components/Feeds/Feeds.js:
```diff
@@ -0,0 +1,188 @@
+import React from "react";
+import "./Feeds.scss";
```
import 순서 수정해주세요!

joonsikyang commented on src/pages/yejikang/Main/Main.js:
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import "./Main.scss";
```
.scss import는 마지막에!

joonsikyang reviewed (commented):
잘해주셨네요 예지님🙃 리뷰 확인해주세요.

> 댓글 추가하는 기능을 'Comment Component' 로 분리했습니다.(이것은 feeds.js 에서 사용 되는 컴퍼넌트입니다.) 스타일은 따로 분리해서 작성하지 않았습니다. Feeds.js 에서도 쓰이고 , Comment.js에서도 쓰이는 것이 있어서 분리를 안했습니다. ex) .heart , .fillheart
혹시 중복이 되더라도 따로 작성을 해야하는지 궁금합니다.

- 컴포넌트 파일을 분리했다면 스타일 파일도 분리하는게 유지보수 측면에서 훨씬 좋습니다! 분리해보시면 좋을 것 같습니다. 해보시고 다른 컴포넌트에서 다른 스타일 주는 것에 대해서도 고민해보시기 바랍니다 :)

Kangyejiii committed:
Refactor : 코드리뷰 반영(import 순서/ 컴포넌트 분리)
- import 순서 정리
- comment component scss 파일 생성 (feeds 컴포넌트와  스타일 분리 )

Kangyejiii commented on src/pages/yejikang/Main/Components/Aside/Aside.js:
```diff
@@ -1,6 +1,6 @@
 import React from "react";
-import "./Aside.scss";
 import { Link } from "react-router-dom";
+import "./Aside.scss";
```
refactoring) import 순서는 보통 패키지 > 컴포넌트 > 변수/이미지 > 마지막으로 .scss 파일 순서!

Kangyejiii commented on src/pages/yejikang/Main/Components/Feeds/Feeds.scss:
```diff
@@ -0,0 +1,222 @@
+						width: 557.77px;
+						height: 18px;
+					}
+					button {
+						background-color: inherit;
+						color: $blue;
+						font-weight: 600;
+
+						&:disabled {
+							opacity: 0.7;
+							pointer-events: none;
+						}
```
📌 sass 의 & 연산자 사용하기

Kangyejiii commented on src/Components/Nav/Nav.scss:
```diff
@@ -1,32 +1,28 @@
-		display: block;
-	}
+	background-color: $white;
+	border-bottom: 1px solid $border-grey;
```
📌 $변수 사용하기, 팀원들이 같이 쓰는 색깔 common.scss에 작성

dipsiiiiiiiiii commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+@import "../../styles/common.scss";
+
+@mixin center-fixed {
+	display: flex;
+	justify-content: center;
+	position: fixed;
```
mixin 쓰신 거 너무 좋네요!

dipsiiiiiiiiii commented on src/Components/Nav/Nav.scss:
```diff
@@ -0,0 +1,67 @@
+	border-bottom: 1px solid $border-grey;
+
+	nav {
+		@include centerSpaceBetween;
```
탭이 너무 많이 들어가는 것 같애요! vscode 설정 변경해주세요!

dipsiiiiiiiiii commented on src/pages/yejikang/Login/Login.js:
```diff
@@ -0,0 +1,86 @@
+  };
+
+  handleInputChange = (e) => {
+    this.setState({ [e.target.name]: e.target.value });
```
비구조화 가르쳐 드릴게요!

```
const { name, value } = e.target;
this.setState({[name] : value})
```

dipsiiiiiiiiii commented on src/pages/yejikang/Login/Login.js:
```diff
@@ -0,0 +1,86 @@
+    fetch("http://10.58.4.11:8000/user/signin", {
+      method: "POST",
+      body: JSON.stringify({
+        email: this.state.id,
+        password: this.state.pw,
```
여기서도 비구조화를 할 수 있는데 만약에 key와 value가 똑같으면 하나로 합칠 수 있어요!
예를들어서 
```
{
   id: id,
   pw: pw,
}
```
라고 한다면
```
{
   id,
   pw
}
```

이렇게 줄일 수 있습니다 :)

dipsiiiiiiiiii commented on src/pages/yejikang/Login/Login.js:
```diff
@@ -0,0 +1,86 @@
+
+  render() {
+    const { id, pw } = this.state;
+    const enabled = id.includes("@") && pw.length >= 5;
```
변수로 빼서 관리하신 거 너무 좋네요 !

dipsiiiiiiiiii commented on src/pages/yejikang/Main/Components/Aside/Aside.js:
```diff
@@ -0,0 +1,169 @@
+                </div>
+                <button>팔로우</button>
+              </div>
+              <div className="listItem">
+                <Link className="listPicture" to="/">
+                  <img
+                    alt="mynameis20님의 프로필"
+                    className="profileImg"
+                    src="https://scontent-gmp1-1.cdninstagram.com/v/t51.2885-19/s150x150/29403423_2018860301734687_8265582011880570880_n.jpg?_nc_ht=scontent-gmp1-1.cdninstagram.com&amp;_nc_ohc=m0CvUVd59rEAX-79ddt&amp;oh=593dd7d6cb1922837ce85c1741bd7b41&amp;oe=5F515228"
+                  />
+                </Link>
+                <div className="listDescription">
+                  <Link to="/" className="suggestionsId">
+                    mynameis20
+                  </Link>
+                  <div>
+                    <span className="comment">회원님을 팔로우합니다.</span>
+                  </div>
+                </div>
+                <button>팔로우</button>
+              </div>
```
이거 하나가 반복되서 컴포넌트로 따로 빼고 map을 사용하실 수도 있습니다!

Kangyejiii replied:
> 이거 하나가 반복되서 컴포넌트로 따로 빼고 map을 사용하실 수도 있습니다!

네 수정했습니다!

dipsiiiiiiiiii commented on src/pages/yejikang/Main/Components/Feeds/Feeds.js:
```diff
@@ -0,0 +1,188 @@
+        <section className="feedSection">
+          <div className="storyContainer">
+            <ul>
+              <li className="storyItem">
+                <Link to="/" className="storyCircle">
+                  <img
+                    alt="__님의 프로필 사진"
+                    className="profileImg"
+                    src="images/yejikang/me.jpg"
+                  />
+                </Link>
+                <span className="storyId">__hi</span>
+              </li>
```
이것도 반복되서 컴포넌트로 따로 빼고 map 사용하실 수 있을 거 같애요!

dipsiiiiiiiiii commented on src/pages/yejikang/Main/Components/Feeds/Feeds.js:
```diff
@@ -0,0 +1,188 @@
+                    key={index}
+                    id={this.state.id}
+                    text={comment}
+                  ></Comment>
```
self closing 써주세요!

dipsiiiiiiiiii commented on src/pages/yejikang/Main/Main.js:
```diff
@@ -0,0 +1,22 @@
+class Main extends React.Component {
+  render() {
+    return (
+      <div className="Main">
+        <Nav />
+        <div className="Section">
+          <Feeds />
+          <Aside />
+        </div>
+      </div>
```
너무 깔끔합니다!
nav아래의 div 태그에서 main 태그 써주셨으면 더 좋을 것 같네요!

dipsiiiiiiiiii reviewed (commented):
너무 잘하시네요! 리뷰 바로 반영해주시면 머지하겠습니다!

Kangyejiii committed:
[Refactor] 중복된 코드 Storys Component로 분리
- Feeds.js에서 스토리 코드 부분 Component로 분리
- Storys Component 생성
- Storys.scss 생성 ,( Feeds.scss에서 분리 )

Kangyejiii committed:
Refactor : Aside의 SuggestionList를  Component로 분리
- data/yejikangMock 안에 추천리스트 받는 Mockdata 추가
- fetch 함수가 있는 별도의 파일 추가(RemoteDataSource)
- SuggestionListItem Component 추가
- 관련된 스타일 suggestionListItem. scss 파일 추가 (Aside.scss에서
  스타일 분리)

Kangyejiii committed:
Refactor : destructuring 하기 ,main 불필요한 스타일 제거
- 비구조화(ES6)를 활용한 코드 수정
- Main.scss 불필요한 스타일 제거

Kangyejiii committed:
Refactor : destructuring 추가 작업
- SuggestionListItem.js 비구조화.

Kangyejiii committed:
Refactor : Login 필요없는 div 정리
- login 레이아웃 정리
- 필요없는 div 제거
- login.scss 수정
- 글씨크기 수정

Kangyejiii committed:
Refactor : storty  mockdata 추가 및 적용,시맨틱태그 사용
- story mock data 추가 및 적용
- story scroll 숨기기 기능 추가
- Main.js 에서 시맨틱 태그로 수정

Kangyejiii committed:
Refactor : [Aside] 중복요소 , 시맨틱태그 ,[Feeds] spread 연산자
- Aside / 하단 도움말: 배열 생성 후 map()
- Feeds / spread 연산자 사용
- 시맨틱 태그 사용

Kangyejiii committed:
Refactor : 불필요한 div,주석 제거
- Nav,Aside  불필요한 div 제거
- 주석 제거

Kangyejiii committed:
indent test
